emcc.py

```diff
@@ -904,11 +904,16 @@ def add_link_flag(i, f):
 
       link_flags.append((i, f))
 
+    # variable to check whether stdin contains code
+    pipe_stdin = False
     # find input files with a simple heuristic. we should really analyze
     # based on a full understanding of gcc params, right now we just assume that
     # what is left contains no more |-x OPT| things
     for i in range(len(newargs)):
       arg = newargs[i]
+      if arg == '-x' and not sys.stdin.isatty():
+        pipe_stdin = True
```

Comment:
I believe gcc and clang only try to compile from stdin when you pass `-` on its own.     In this case '-' acts as just another input file.

Its even possible to compile several sources files *and* stdin at the same time: `clang -c foo.c bar.c -` for example.